sklearnex/linear_model/incremental_linear.py

```diff
@@ -150,30 +151,32 @@ def _onedal_predict(self, X, queue=None):
     def _onedal_partial_fit(self, X, y, queue=None):
         first_pass = not hasattr(self, "n_samples_seen_") or self.n_samples_seen_ == 0
 
-        if sklearn_check_version("1.2"):
-            self._validate_params()
-
-        if sklearn_check_version("1.0"):
-            X, y = self._validate_data(
-                X,
-                y,
-                dtype=[np.float64, np.float32],
-                reset=first_pass,
-                copy=self.copy_X,
-                multi_output=True,
-            )
-        else:
-            X = check_array(
-                X,
-                dtype=[np.float64, np.float32],
-                copy=self.copy_X,
-            )
-            y = check_array(
-                y,
-                dtype=[np.float64, np.float32],
-                copy=False,
-                ensure_2d=False,
-            )
+        if not hasattr(self, "batch_size_"):
```

Comment:
why is condition needed?

Reply:
yeah, good catch.  batch_size is the only way to tell that onedal_partial_fit was called via `fit` and not `partial_fit`. If it was called from `partial_fit` data validation needs to occur on the input, if it was called from `fit` it has already been done.

Reply:
I changed it to match the convention in sklearn's IncrementalPCA which uses a check_input kwarg.